### PR TITLE
PMM-1834: Filter out empty comments and "# explain:" for MariaDB.

### PR DESCRIPTION
**Pull request by arvenil**
* [PMM-1834: Slow log parse error: explain as part of a fingerprint](https://jira.percona.com/browse/PMM-1834)
  This slow log:
  ```
  mysqld, Version: 10.2.13-MariaDB-10.2.13+maria~jessie-log (mariadb.org binary distribution). started with:
  Tcp port: 0  Unix socket: /var/run/mysqld/mysqld.sock
  Time                 Id Command    Argument
  # Time: 180214 16:18:07
  # User@Host: root[root] @ localhost []
  # Thread_id: 8  Schema:   QC_hit: No
  # Query_time: 0.000277  Lock_time: 0.000000  Rows_sent: 1  Rows_examined: 0
  # Rows_affected: 0
  #
  # explain: id	select_type	table	type	possible_keys	key	key_len	ref	rows	r_rows	filtered	r_filtered	Extra
  # explain: 1	SIMPLE	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	  NULL	NULL	No tables used
  #
  SET timestamp=1518625087;
  SELECT 1;
  ```
  was resulting in this query:
  ```
  #\n# explain: id\tselect_type\ttable\ttype\tpossible_keys\tkey\tkey_len\tref\trows\tr_rows\tfiltered\tr_filtered\tExtra\n# explain: 1\tSIMPLE\tNULL\tNULL\tNULL\tNULL\tNULL\tNULL\tNULL\tNULL\tNULL\tNULL\tNo tables used\n#\nSELECT 1
  ```
  instead of just:
  ```
  SELECT 1
  ```
* `-update` flag to update test data without hassle.

### CODE REVIEW

File: event/aggregator_test.go [hunk 1]
```diff
@@ -60,7 +55,18 @@ func aggregateSlowLog(input, output string, utcOffset time.Duration, examples bo
 		a.AddEvent(e, id, f)
 	}
 	got := a.Finalize()
-	gotJson, err := json.Marshal(got)
+	gotJson, err := json.MarshalIndent(got, "", "  ")
+	if err != nil {
+		l.Fatal(err)
```
Comment (AlekSi) [inline event/aggregator_test.go]: I wonder if we should stop mixing `testing` and testify in the same file at least for new code?

File: event/aggregator_test.go [hunk 2]
```diff
@@ -183,3 +189,8 @@ func TestOutlierSlow025(t *testing.T) {
 	got, expect := aggregateSlowLog("slow025.log", "slow025.json", 0, true)
 	assert.JSONEq(t, expect, got)
 }
+
+func TestMariaDB102WithExplain(t *testing.T) {
+	got, expect := aggregateSlowLog("mariadb102-with-explain.log", "mariadb102-with-explain.json", 0, true)
+	assert.JSONEq(t, expect, got)
+}
```
Comment (AlekSi) [inline event/aggregator_test.go]: Newline, please.

File: test/test.go [hunk 3]
```diff
@@ -4,8 +4,11 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"flag"
 )
 
+var Update = flag.Bool("update", false, "update .golden files")
```
Comment (AlekSi) [inline test/test.go]: But they are not `.golden`. 😄 I think the description could be better.

File: test/test.go [hunk 4]
```diff
@@ -4,8 +4,11 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"flag"
```
Comment (AlekSi) [inline test/test.go]: Looks like `gofmt` wasn't applied.

File: event/aggregator_test.go (hunk 1 above)
Comment (arvenil) [reply event/aggregator_test.go]: Not sure what you mean here, `l.Fatal(err)` refers here to `log` package, beside it's an old code, I've added indent only.

**Comment by arvenil** [comment]
Everything else should be fixed.

File: event/aggregator_test.go (hunk 1 above)
Comment (AlekSi) [reply event/aggregator_test.go]: Ah, that helper does not accept `testing.T`. Well, ok.

**Pull request merged**